From f5af0d46904125f0c4c874b462c9f44d4688a528 Mon Sep 17 00:00:00 2001
From: Andreas Sturmlechner <asturm@gentoo.org>
Date: Wed, 19 Dec 2018 14:21:13 +0100
Subject: [PATCH] Use KF5_MIN_VERSION instead of KF5_VERSION

Summary:
KF5_VERSION is overridden by first KF5 find_package version found result, which may be higher
than other installed Frameworks versions at the time, leading to cmake error.

Reviewers: #kde_pim, dvratil

Reviewed By: #kde_pim, dvratil

Subscribers: dvratil, kde-pim

Tags: #kde_pim

Differential Revision: https://phabricator.kde.org/D17684
---
 CMakeLists.txt | 10 ++++------
 1 file changed, 4 insertions(+), 6 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index f54a17f..708a266 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -4,9 +4,9 @@ set(PIM_VERSION "5.10.0")
 project(AkonadiSearch VERSION ${PIM_VERSION})
 
 # ECM setup
-set(KF5_VERSION "5.51.0")
+set(KF5_MIN_VERSION "5.51.0")
 
-find_package(ECM ${KF5_VERSION} CONFIG REQUIRED)
+find_package(ECM ${KF5_MIN_VERSION} CONFIG REQUIRED)
 set(CMAKE_MODULE_PATH ${ECM_MODULE_PATH})
 
 include(GenerateExportHeader)
@@ -39,9 +39,9 @@ set(KMIME_LIB_VERSION "5.10.0")
 set(QT_REQUIRED_VERSION "5.9.0")
 
 find_package(Qt5 ${QT_REQUIRED_VERSION} CONFIG REQUIRED Core Test)
-find_package(KF5 ${KF5_VERSION} REQUIRED COMPONENTS I18n Config Crash KCMUtils)
+find_package(KF5 ${KF5_MIN_VERSION} REQUIRED COMPONENTS I18n Config Crash DBusAddons KCMUtils KIO)
 if (AKONADISEARCH_ENABLE_KRUNNER_RUNNER)
-    find_package(KF5Runner ${KF5_VERSION} REQUIRED)
+    find_package(KF5Runner ${KF5_MIN_VERSION} REQUIRED)
 endif()
 
 
@@ -57,8 +57,6 @@ find_package(KF5Contacts ${KCONTACTS_LIB_VERSION} CONFIG REQUIRED)
 find_package(KF5Mime ${KMIME_LIB_VERSION} CONFIG REQUIRED)
 find_package(KF5AkonadiMime ${AKONADI_MIMELIB_VERSION} CONFIG REQUIRED)
 find_package(KF5CalendarCore ${KCALENDARCORE_LIB_VERSION} CONFIG REQUIRED)
-find_package(KF5KIO ${KF5_VERSION} CONFIG REQUIRED)
-find_package(KF5DBusAddons ${KF5_VERSION} CONFIG REQUIRED)
 
 ecm_setup_version(PROJECT
     VARIABLE_PREFIX AKONADISEARCH
-- 
2.20.1

